app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultScreen.kt

```diff
@@ -0,0 +1,181 @@
+import android.view.View
+import android.widget.RelativeLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.xwray.groupie.GroupAdapter
+import com.xwray.groupie.ViewHolder
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.ofType
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.patient_search_view.view.*
+import kotlinx.android.synthetic.main.screen_shortcode_search_result.view.*
+import org.simple.clinic.R
+import org.simple.clinic.ViewControllerBinding
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.bp.BloodPressureMeasurement
+import org.simple.clinic.facility.FacilityRepository
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.search.PatientSearchScreenKey
+import org.simple.clinic.searchresultsview.PatientSearchResults
+import org.simple.clinic.searchresultsview.SearchResultClicked
+import org.simple.clinic.searchresultsview.SearchResultsItemType
+import org.simple.clinic.summary.OpenIntention
+import org.simple.clinic.summary.PatientSummaryScreenKey
+import org.simple.clinic.text.style.TextAppearanceWithLetterSpacingSpan
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.Truss
+import org.simple.clinic.util.Unicode
+import org.simple.clinic.util.UtcClock
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.hideKeyboard
+import org.threeten.bp.Instant
+import java.util.UUID
+import javax.inject.Inject
+
+class ShortCodeSearchResultScreen(context: Context, attributes: AttributeSet) : RelativeLayout(context, attributes), ShortCodeSearchResultUi {
+
+  @Inject
+  lateinit var uiChangeProducer: ShortCodeSearchResultUiChangeProducer
+
+  @Inject
+  lateinit var utcClock: UtcClock
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @Inject
+  lateinit var patientRepository: PatientRepository
+
+  @Inject
+  lateinit var facilityRepository: FacilityRepository
+
+  @Inject
+  lateinit var userSession: UserSession
+
+  @Inject
+  lateinit var bloodPressureDao: BloodPressureMeasurement.RoomDao
+
+  @Inject
+  lateinit var schedulersProvider: SchedulersProvider
+
+  private lateinit var binding: ViewControllerBinding<UiEvent, ShortCodeSearchResultState, ShortCodeSearchResultUi>
+
+  private val adapter = GroupAdapter<ViewHolder>()
+
+  private val downstreamEvents = PublishSubject.create<UiEvent>()
+
+  private lateinit var disposable: Disposable
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) {
+      return
+    }
+    hideKeyboard()
+
+    TheActivity.component.inject(this)
+
+    val screenKey = screenRouter.key<ShortCodeSearchResultScreenKey>(this)
+    setupToolBar(screenKey)
+    setupScreen()
+
+    val uiStateProducer = ShortCodeSearchResultStateProducer(
+        shortCode = screenKey.shortCode,
```

Comment:
The only reason we do too many field injections in this call is due to this dynamic dependency. If we create a submodule that accepts this value as a parameter. we should be able to get away with,

```
@Inject
lateinit var uiStateProducer: ShortCodeSearchResultStateProducer

@Inject
lateinit var uiChangeProducer: ShortCodeSearchResultUiChangeProducer
```